### PR TITLE
Oh god. Oh duck. Adds dupe duck.

### PR DESCRIPTION
**Pull request by DrSnips**
[honk] [ohgoditsloose]
Have you ever wanted a rubber duck that creates a duplicate of itself, launches it in a random direction, and the cycle repeats? Neither have I but I made this cursed item. It creates about 50 of them and then stops for reasons unknown to me.
Webm showing off it's raw power.
https://webmshare.com/YzGVw

:cl:
 * rscadd: Added the dupe duck. God help us all.

### CODE REVIEW

**Comment by kilozombie** [comment]
>there's nothing in the code that stops it, the gods just inexplicably cut the curse short

I'm scared

**Comment by Kharkov72** [comment]
oh god, oh fuck
post webm

**Comment by DrSnips** [comment]
op webm soon

**Comment by DrSnips** [comment]
OP updated with a webm.

**Comment by ghost** [comment]
what have you unleashed upon this accursed land

**Comment by aceedex** [comment]
but why?

**Comment by tuypo1** [comment]
I feel like this would be more interesting if there was more of a delay between each dupe.

**Comment by DrSnips** [comment]
@aceedex  https://youtu.be/VPpIjhtgGj0

**Comment by plosky1** [comment]
Is it possible to make it double each time instead of just vomiting out 50? Would also be cool if it has the same effect as the teleporting duck with sparks.

**Comment by Exxion** [comment]
Presumably the reason it stops is that it hits some kind of recursion limit
This is a terrible thing to rely on for obvious reasons
Also this is just a terrible idea in general
You're adding something to the code knowing that it should never be spawned
It's completely pointless

**Comment by tuypo1** [comment]
>knowing that it should never be spawned

Fuck that shit, throw it in the syndie uplink for 5 credits under pointless badasery, the category is way too empty

**Comment by PaulRitter** [comment]
\>It creates about 50 of them and then stops for reasons unknown to me.
then create a fixed limit, dont do that

File: code/game/objects/items/misc_items.dm [hunk 1]
```diff
@@ -70,4 +70,28 @@
...
-	qdel(src)
\ No newline at end of file
+	qdel(src)
+
+//////////Cursed items//////////
+
+//Namely the Gemino Curse. God help anyone who spawns this.
+/obj/item/weapon/bikehorn/rubberducky/dupeduck
+	desc = "Oh God. Oh Duck."
+	var/willdupe = FALSE
```
Comment (Kurfursten) [inline code/game/objects/items/misc_items.dm]: var defined but unused?? you don't check this value anywhere

**Comment by tuypo1** [comment]
I agree with Paulritter, I'm no coder but just using the unexplained limit instead of making a hard one seems like a bad idea.

**Comment by DrSnips** [comment]
The virgin coded hard limit VS The Chad fabric of reality allowance limit.

**Comment by Exxion** [comment]
That's not optional if for some unknowable reason you actually want this merged

**Comment by DrSnips** [comment]
Adding a delay to them means they need to be added to the processing objects and this would likely cause even worse performance issues. So I won't be adding a delay unless you give me a good reason.

**Comment by PaulRitter** [comment]
@DrSnips please read my comment again

**Comment by PaulRitter** [comment]
also adding a delay would not mean you'd need to add it to the processing objs, you could just **spawn with delay** the code where you spawn and throw it

but just so we are on the same page, i was previously talking about adding a hardcoded limit instead of going "huh stopped there yeah". I assume exxion was referring to that too

**Comment by Durgen** [comment]
\>doesn't crash the server
lame

**Comment by Exxion** [comment]
@DrSnips First, no, a delay does not require you to add it to `processing_objects`
Second, the thing I said was not optional was a limit to the number of duplications

**Comment by Blithering** [comment]
first there were teleducks, now we have recursive ducks. Anomalous Duck Handling Department when?

**Comment by FudgePucker** [comment]
Merge when?

**Comment by tuypo1** [comment]
When the hard limit is added

**Comment by SonixApache** [comment]
add the limit or close this dumb shit

**Comment by Kurfursten** [comment]
Even adminbus stuff shouldn't run on intentional badcode. C'mon.

**Comment by FudgePucker** [comment]
Using the engines limit instead of a hard limit reduces bloat :^)

**Comment by DrSnips** [comment]
It seems that the collabs are still too afraid of how powerful my code is. Maybe next time I code a random shitpost I saw it will be merged but sadly this one will not. Tune in next time for more !FUN! code.

**Comment by aceedex** [comment]
Collabs FEAR the average contributor

**Comment by Kurfursten** [comment]
You have enough voting support, just fix it. It's an admintoy anyway.

I for one will change my vote if it doesn't run on shitcode.

**Pull request closed without merging**